src/features/account/components/profile-form.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { User as AuthUser } from "lucia";
+import { useActionState } from "react";
+import { FieldError } from "@/components/form/field-error";
+import { Form } from "@/components/form/form";
+import { SubmitButton } from "@/components/form/submit-button";
+import { EMPTY_ACTION_STATE } from "@/components/form/utils/to-action-state";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { updateProfile } from "../actions/update-profile";
+
+type ProfileFormProps = {
+  user: AuthUser;
+};
+
+const ProfileForm = ({ user }: ProfileFormProps) => {
+  const [actionState, action] = useActionState(
+    updateProfile.bind(null),
+    EMPTY_ACTION_STATE
+  );
+
+  return (
+    <Form action={action} actionState={actionState}>
+      <Label htmlFor="username">Username</Label>
+      <Input
+        id="username"
+        name="username"
+        type="text"
+        defaultValue={
+          (actionState.payload?.get("username") as string) ?? user.username
+        }
+      />
+      <FieldError actionState={actionState} name="username" />
+
+      <Label htmlFor="email">Email</Label>
```

Comment:
Updating E-Mail would technically work this way, but in a real world application, it shouldn't be this straightforward, because usually an email is sent out to this user to confirm the email. Only then the new email would be valid. But this is a thing you will learn in the Software Engineer Journey!

Reply:
Yes, I am aware of this. The exercise was simply to update the database. Additionally, at this stage of the app, we haven't even implemented email verification during registration.